src/redux/reducers/inventory-reducer.js

```diff
@@ -0,0 +1,38 @@
+import { FETCH_INVENTORY_GROUP, FETCH_INVENTORY_GROUP_RESOURCES } from '../action-types';
+import { defaultSettings } from '../../helpers/shared/pagination';
+
+export const inventoryGroupsInitialState = {
+  isLoading: false,
+  resourceTypes: {
+    data: [],
+    meta: defaultSettings,
+  },
+  resources: {},
+  loadingResources: 0,
+};
+
+const setLoadingState = (state) => ({ ...state, isLoading: true });
+
+const setResourceTypes = (state, { payload }) => ({
+  ...state,
+  resourceTypes: {
+    ...state.resourceTypes,
+    data: payload.results,
+  },
+  isLoading: false,
+});
+
+const setResource = (state, { payload }) => ({
+  ...state,
+  resources: { ...state.resources, [payload.links.first.split('/')[5]]: payload.data.filter(({ value }) => value !== null) },
+  loadingResources: state.loadingResources - 1,
+});
+
+const setResourceLoading = (state) => ({ ...state, loadingResources: state.loadingResources + 1 });
```

Comment:
We add to the number of loading resources and when it reaches 0 it means none are loaded?